### ISSUE TITLE
Selection paintbrush feature

### ISSUE THREAD

**Issue by KashouC** [aseprite/aseprite#1787]
This is something I've wanted in quite a few programs. What I'm getting at is basically just the pen tool, but instead of painting paint, it paints out selection just like the lasso or marquee tool. I think it would be very helpful in a lot of situations, especially with lots of angles and such, and while the lasso works it's kind of clunky imo.

**Comment by dacap** [comment]
Hi @KashouC, actually something like that is possible but there is no easy way to do it. You have to modify the `data/gui.xml` file manually. After the `<tool id="pencil" ...>` [section](https://github.com/aseprite/aseprite/blob/master/data/gui.xml#L1087) you might try to add something like this:
```
      <tool id="sel_pencil"
            text="Selection Pencil"
            ink="selection"
            controller="freehand"
            pointshape="brush"
            intertwine="as_lines"
            tracepolicy="accumulate" />
```
The bad side that I see right now is that the brush size is not shown in this case, so it's something to be fixed in the future, also I might add a new "Ink" mode for this kind of behavior:

![screen shot 2018-07-09 at 20 09 14](https://user-images.githubusercontent.com/39654/42480569-911f4038-83b4-11e8-9a54-b9adb9f0fd59.png)

More info about the internals here: https://dev.aseprite.org/2015/06/06/the-tool-loop/

**Comment by KashouC** [comment]
Mmm, interesting. I'll check it out, when it's not 2 in the morning. My only issue with an ink is that I'd like to be able to keybind it directly, but now that I look it up it's entirely possible to do just that.

**Comment by KashouC** [comment]
Yeah this is pretty much exactly what I had in mind. Not being able to change the brush size kind of leaves it barely useful though but it's very close.
